### DISCUSSION TITLE
Symfony 6.2 infinite dependency-injection loop error

### DISCUSSION THREAD

**Discussion by hanishsingla** [symfony/symfony#48515] [Q&A] [answered]
I am trying to update my project from Symfony 6.1 to 6.2, but after composer finishes installing updated bundles, console starts throwing up following error.

```
#message: "Xdebug has detected a possible infinite loop, and aborted your script with a stack depth of '256' frames"
#code: 0
#file: "{projectPath}\vendor\symfony\dependency-injection\Compiler\ResolveNoPreloadPass.php"
#line: 61
```

These are files, which are causing this error

```
    !!      {projectPath}\vendor\symfony\dependency-injection\Compiler\AbstractRecursivePass.php:89 { …}
    !!      {projectPath}\vendor\symfony\dependency-injection\Compiler\ResolveNoPreloadPass.php:86 { …}
    !!      {projectPath}\vendor\symfony\dependency-injection\Compiler\ResolveNoPreloadPass.php:68 { …}
    !!      {projectPath}\vendor\symfony\dependency-injection\Compiler\AbstractRecursivePass.php:80 { …}
    !!      {projectPath}\vendor\symfony\dependency-injection\Compiler\ResolveNoPreloadPass.php:75 { …}
```

Any symfony command (e.g. php bin/console assets:install public or php bin/console cache:clear) throws same error

I even tried to clear the cache suspecting some old class is causing this issue, but no success. All recipes are up to date. Even removing vendor folder and re-installing 6.2 bundles hasn't fixed this issue.

Has anyone experienced this issue while updating Symfony? I am really not sure where to start looking up for the solution

For more info about the error, please refer to this SO question

https://stackoverflow.com/questions/74652270/symfony-6-2-infinite-dependency-injection-loop

**Comment by stof** [comment, marked as answer]
Does it also happen if you increase the max recursion from 256 to 1000 ? In my experience, it is possible to have some deep recursion depending on your service graph, but it is still not infinite.

**Reply by hanishsingla** [reply]
Thanks for this solution!! I found out that in new version, WAMP had enabled xDebug automatically, which was also a reason why env was suddenly so slow. I disabled the xDebug and error is gone

**Reply by andrew-demb** [reply]
Hello @stof. We are having the same issue with increased nesting level and the same stack trace.
Is there any way not to increase xdebug max nesting level? 

Why `symfony/dependency-injection` since 6.2 requires more deep processing?

**Comment by curry684** [comment]
We're also suddenly running into this in a project that uses API Platform. Container won't compile with 275, will with 300.

I think @derickr should consider raising the default from 256 to 512 if standard frameworks with standard libraries start running into this.

**Comment by derickr** [comment]
Stop making your code so complicated :-þ — I saw you filed a ticket, @curry684, I'll get that into Xdebug 3.3.

**Reply by derickr** [reply]
@curry684 I can't actually find the ticket/bug report, did you file it at https://bugs.xdebug.org?

**Reply by curry684** [reply]
@derickr nope sorry made a PR on it right away here: https://github.com/xdebug/xdebug/pull/910

Should be complete from what I saw (but am completely noob to the Xdebug codebase).
